test/test_operations.py

```diff
@@ -2835,6 +2836,21 @@ def test_dropout(self):
     self.assertTrue(same_output,
                     f"in fwd {model.to_save[0]}, in bwd {model.to_save[1]}")
 
+  def test_opt_barrier(self):
+    device = xm.xla_device()
+    model = SimpleModelWithDropout().to(device)
+    model = checkpoint_module(model)
+    _input = torch.randn(128, 128, requires_grad=True)
+    _input = _input.to(device)
+    output = model(_input)
+    output = torch.sum(output)
+    output.backward()
+
+    hlo = torch_xla._XLAC._get_xla_tensors_hlo([model.x.weight.grad])
```

Comment:
didn't your change include the `weight` and `buffer`, but the test seems to try to test the HLO of the `weight.grad`

Reply:
That's the weight to get the full hlo. As long as the opt-barrier contains all tensors, it's fine.